airbyte-integrations/connectors/source-waids-oscarpro/source_waids_oscarpro/source.py

```diff
@@ -209,8 +208,8 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         """
         # TODO remove the authenticator if not required.
 
-        # TODO --> The values needed for Authentication are on the config var
-
-
-        auth = TokenAuthenticator(token="api_key")  # Oauth2Authenticator is also available if you need oauth support
-        return [Customers(authenticator=auth), Employees(authenticator=auth)]
+        oscar_pro_authentication = OscarProAuthentication(config)
+        token = oscar_pro_authentication.get_access_token()
+        auth = TokenAuthenticator(token)  # Oauth2Authenticator is also available if you need oauth support
+        
+        return [Customers(config=config, authenticator=auth), Employees(config=config, authenticator=auth)]
```

Comment:
This is where the token get's injected into the stream.

Basically, we're passing in the configs' Mapping into the stream (needed so we can get the `url_base` as a variable instead of hardcoded

Then authenticator is the `auth` type which is fed by the token we generated before hand